### ISSUE TITLE
Improvement on lineage with airflow dags

### ISSUE THREAD

**Issue by ggnogueira** [open-metadata/OpenMetadata#12951]
**Is your feature request related to a problem? Please describe.**
Usually, airflow DAGs can have several tasks, where each task originates a table in your datalake. For example, a task named (A) reads data from Kafka source and saves it in the raw layer as a bronze table. Task (B) reads the bronze table, performs data transformations and returns the silver table. Finally, task C reads the silver data, performs some aggregations and returns a gold.

In short, allow in the edge pipeline editor make reference to a DAG task.

**Describe the solution you'd like**
Allow in the edge pipeline editor make reference to a DAG task.

**Comment by KsM00000** [comment]
Yeah, I would like that too.
Our tasks (usually stored procedures) work with a decent number of tables. No division into tasks... so our pipeline's lineages mostly looks like this :/
![image](https://github.com/open-metadata/OpenMetadata/assets/79855848/bca44243-27d7-4642-bd46-d5d7820e79a1)
In my opinion, if the tables, when they reach a certain number, were hidden, breaking up according to the schemes, it would be great.

**Comment by moritzsanne** [comment]
@chirag-madlani I don't believe @ggnogueira initial request was addressed by [Lineage layers #15940](https://github.com/open-metadata/OpenMetadata/pull/15940). We just tested lineage extraction from airflow, and edges still only refer to the DAG not the actual task. Or am I missing something?
